REFERENCE.md

```diff
@@ -658,3 +742,41 @@ Which daemon file in which to write the user.  snmpd or snmptrapd
 
 Default value: 'snmpd'
 
+## Functions
+
+### snmp::snmpv3_usm_hash
+
+Type: Ruby 4.x API
+
+snmpv3_usm_hash.rb --- Calculate SNMPv3 USM hash for a passphrase
+
+#### `snmp::snmpv3_usm_hash(String $authtype, String $engine, String $passphrase, Optional[Integer] $bits)`
+
+snmpv3_usm_hash.rb --- Calculate SNMPv3 USM hash for a passphrase
+
+Returns: `String` The calculated hash.
+
+##### `authtype`
+
+Data type: `String`
+
+The authentication type to calculate. This must either be
```

Comment:
puppet-strings isn't very good with function docs.  I think this function is private, so have opened https://github.com/voxpupuli/puppet-snmp/pull/186

@smoeding I see you contributed this function.  Am I right in thinking it's private? (Comments on the original PR suggest to me it is)

Reply:
There might maybe be a time when this might be handy outside of the module. But currently I would think it is safe to keep it private to the module.